src/query/src/range_select/plan.rs

```diff
@@ -118,18 +127,63 @@ impl RangeSelect {
         let by_fields =
             exprlist_to_fields(by.iter().collect::<Vec<_>>(), &input).context(DataFusionSnafu)?;
         fields.extend(by_fields.clone());
-        let schema = DFSchema::new_with_metadata(fields, input.schema().metadata().clone())
-            .context(DataFusionSnafu)?;
-        let by_schema = DFSchema::new_with_metadata(by_fields, input.schema().metadata().clone())
-            .context(DataFusionSnafu)?;
+        let schema_before_project = Arc::new(
+            DFSchema::new_with_metadata(fields, input.schema().metadata().clone())
+                .context(DataFusionSnafu)?,
+        );
+        let by_schema = Arc::new(
+            DFSchema::new_with_metadata(by_fields, input.schema().metadata().clone())
+                .context(DataFusionSnafu)?,
+        );
+        // If the result of the project plan happens to be the schema of the range plan, no project plan is required
+        // that need project is identical to range plan schema.
+        // 1. all exprs in project must belong to range schema
+        // 2. range schema and project exprs must have same size
+        let range_schema_fields = schema_before_project.fields();
+        let schema_project = if projection_expr.len() == range_schema_fields.len() {
+            let mut project_map = Vec::with_capacity(projection_expr.len());
+            for project_expr in projection_expr {
+                if let Expr::Column(column) = project_expr {
+                    for (i, v) in range_schema_fields.iter().enumerate() {
+                        if v.qualifier() == column.relation.as_ref() && v.name().eq(&column.name) {
+                            project_map.push(i);
+                            break;
+                        }
+                    }
+                } else {
+                    break;
+                }
+            }
+            if project_map.len() == projection_expr.len() {
+                Some(project_map)
+            } else {
+                None
+            }
+        } else {
+            None
+        };
```

Comment:
non-blocking: it would be better if we can simplify this `if`s...

Reply:
👍

Reply:
@Taylor-lagrange Hi, will u do this refactor? If not, I'll merge this PR.

Reply:
Give me some time to refactor this

Reply:
Done @killme2008